File: lab2plus/vector2.cpp
#include "vector2.h"
#include <cmath>

// Реалізація конструктора за замовчуванням
vector2::vector2() : x(0.0), y(0.0) {}

// Реалізація конструктора з параметрами
vector2::vector2(double x, double y) : x(x), y(y) {}

// Реалізація конструктора копіювання
vector2::vector2(const vector2& other) : x(other.x), y(other.y) {}

// Реалізація деструктора
vector2::~vector2() {}

double vector2::getX() const { return x; }
double vector2::getY() const { return y; }

// Обчислення полярних координат
double vector2::getLength() const {
    return std::sqrt(x * x + y * y);
}

double vector2::getAngle() const {
    return std::atan2(y, x);
}

File: Vector2.cs
using System;

namespace vectorlibrary
{
    public class vector2
    {
        private double _x;
        private double _y;

        // Конструктор за замовчуванням
        public vector2()
        {
            _x = 0.0;
            _y = 0.0;
        }

        // Конструктор з параметрами
        public vector2(double x, double y)
        {
            _x = x;
            _y = y;
        }

        // Конструктор копіювання
        public vector2(vector2 other)
        {
            _x = other._x;
            _y = other._y;
        }

        // Деструктор
        ~vector2()
        {
        }

        public double X { get { return _x; } }
        public double Y { get { return _y; } }

        public double GetLength()
        {
            return Math.Sqrt(_x * _x + _y * _y);
        }

        public double GetAngle()
        {
            return Math.Atan2(_y, _x);
        }
    }
}